src/test/java/com/artipie/RpPermissionsTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie;
+
+import java.io.File;
+import org.cactoos.list.ListOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.AllOf;
+import org.hamcrest.core.IsEqual;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.llorllale.cactoos.matchers.MatcherOf;
+
+/**
+ * Test for {@link RpPermissions}.
+ * @since 0.2
+ * @checkstyle LeftCurlyCheck (500 lines)
+ */
+@Disabled
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+class RpPermissionsTest {
+
+    /**
+     * Test configuration file location.
+     */
+    private static final String CONF_YAML = "src/test/resources/repo_permissions.yml";
+
+    @Test
+    void johnCanDownloadDeployAndDelete() {
+        final String uname = "John";
+        MatcherAssert.assertThat(
+            new RpPermissions(new File(RpPermissionsTest.CONF_YAML)),
+            new AllOf<>(
+                new ListOf<>(
+                    new MatcherOf<>(perm -> { return perm.allowed(uname, "delete"); }),
+                    new MatcherOf<>(perm -> { return perm.allowed(uname, "deploy"); }),
+                    new MatcherOf<>(perm -> { return perm.allowed(uname, "download"); }),
+                    new MatcherOf<>(perm -> { return !perm.allowed(uname, "install"); })
+                )
+            )
+        );
+    }
+
+    @Test
+    void anyoneCanDownload() {
+        MatcherAssert.assertThat(
+            new RpPermissions(new File(RpPermissionsTest.CONF_YAML)).allowed("anyone", "download"),
```

Comment:
@olenagerasimova I believe we should assert `*` instead of `anyone`. `Anyone` can be a valid username.

Reply:
@Vatavuk hm, as far as I understand `*` is special symbol for settings in config file, when `Permissions` implementation should check permissions for real users and actions, so `*` is not supposed to be passed to this implementations. @g4s8 could you please help as here? Can `*` be passed to `Permissions#allowed` as user name?

Reply:
@olenagerasimova no, `*` means "any user" for settings, but it's not a valid username. If we have permission settings like:
```yaml
repo:
  permissions:
    - "*":
      - download
```
then we should allow download for any user (any username)

Reply:
@g4s8 thanks
@Vatavuk please check the answers, do you have any more questions?

Reply:
@olenagerasimova my bad, I've mixed up what are we testing/asserting. No further comments from my side.